src/components/FilterButton.tsx

```diff
@@ -0,0 +1,41 @@
+import React, {MouseEvent} from 'react';
+import "../stylesheets/filterbuttons.scss"
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+import {IconProp} from "@fortawesome/fontawesome-svg-core";
+
+export interface FilterButtonProps {
+    text: string;
+    icon: IconProp;
+    onClick(e: MouseEvent<HTMLElement>): void;
+}
+
+export interface FilterButtonState {
+    selected: boolean;
+}
+
+export class FilterButton extends React.Component<FilterButtonProps, FilterButtonState> {
```

Comment:
Instead of extending the button class, I think it is better to just extend the props and the state. I couldn't find an elegent solution without introducing intermediate generic type as I need to introduce another state (selected) after extending. Let me know what you think.

Reply:
I see. Yea... on further searching, the web says inheritance is not really well supported with React. I think in this case, we can do a composition and nest the Button component inside the FilterButton component?

Reply:
Not sure... I think simplest way is just to reuse the props. The behavior is different in a way that nesting it doesn't make too much sense either.